src/utils/useChromaticDialog.ts

```diff
@@ -0,0 +1,53 @@
+import { useCallback, useEffect, useRef } from "react";
+
+export type DialogEventName = "login";
+export type DialogPayload<T extends DialogEventName> = Record<string, never>;
+export type DialogHandlers = {
+  [Event in DialogEventName]?: (payload: DialogPayload<Event>) => void;
+};
+
+export const useChromaticDialog = (handlers: DialogHandlers = {}) => {
+  const dialog = useRef<Window>();
+  const dialogOrigin = useRef<string>();
+
+  // Close the dialog window when the screen gets unmounted.
+  useEffect(() => () => dialog.current?.close(), []);
+
+  useEffect(() => {
+    const handleMessage = ({ origin, data }: MessageEvent) => {
+      if (origin === dialogOrigin.current) {
+        const { message, ...rest } = data;
+        if (message in handlers) {
+          handlers[message as DialogEventName](rest);
+        }
+      }
+    };
+
+    window.addEventListener("message", handleMessage, false);
```

Comment:
```suggestion
    window.addEventListener("message", handleMessage);
```
`false` is the default for `useCapture`. If you want to be explicit, it's better to use `{ capture: false }` which does the same.